lib/galaxy/util/sanitize_html.py

```diff
@@ -112,17 +114,17 @@ def unknown_starttag(self, tag, attrs):
                 value = value.replace('>', '&gt;').replace('<', '&lt;').replace('"', '&quot;')
                 value = self.bare_ampersand.sub("&amp;", value)
                 # thanks to Kevin Marks for this breathtaking hack to deal with (valid) high-bit attribute values in UTF-8 feeds
-                if isinstance(value, unicode):
+                if isinstance(value, text_type):
                     try:
-                        value = unicode(value, self.encoding)
-                    except:
-                        value = unicode(value, 'iso-8859-1')
-                uattrs.append((unicode(key, self.encoding), value))
+                        value = unicodify(value, self.encoding)
+                    except Exception:
+                        value = unicodify(value, 'iso-8859-1')
```

Comment:
Looking into this it seems to me that the original code in this case is not functional ... I don't see how this would not cause a `TypeError: decoding Unicode is not supported`.

See for comparison the/a possible origin:
https://github.com/basho/planet_riak/blob/master/planet/sanitize.py

Shouldn't this be
```
if isinstance(value, binary):
    try:
        text_type(value, self.encoding)
    except UnicodeDecodeError:
        text_type(value, 'iso-8859-1')
```
?